### ISSUE TITLE
The ethics of this mod

### ISSUE THREAD

**Issue by Thermawrench** [MsRandom/Witchery#1]
Did Emoniph approve of this at all? I'm fairly sure he did not want anyone to mess with his code and assets, that's what the license on it states at least.

**Comment by Gamebuster19901** [comment]
@Emoniph

**Comment by Ruchian** [comment]
Emoniph no longer exists.
Because of this his or her curse licence, especially since curse became twitch, is completely dropped. Because of this Witchery can, according to Twitch be reuploaded and claimed by another user as 'their work'. Mojang refreshed their licence in 2018, meaning the one on the website emoniph once made is outdated. Technically, only the placement of the puzzle pieces and self written pieces would fall under his own licence, but since there is no way to trace emoniph to a real person, anyone can make a twitter account and claim that they're him or a second git account, etc.
It's the concequence of being labeled 'dead' on the internet, I guess. Anyway, due to this, due to his own mistake of inactivity, and anonymousness, his self-established licence cannot protect his work. He may not like it indeed. But in all honesty I'm glad that someone ignored his cries and decided to try and port it. For A. Perhaps it makes Emoniph come back to speak with the community, and B. for the community to enjoy his original work.

Talking about licences, his other mod, intanglible falls completely outside of it. It's not covered by his custom licence at all. So it depends on the distro platform (Twitch), but twitch has no username that is inside the content and equal to the uploaded work, so the licence cannot be formed. Cruelty, but again his own mistake.

**Comment by zabi94** [comment]
> Emoniph no longer exists.
> Because of this his or her curse licence, especially since curse became twitch, is completely dropped.

Completely incorrect. Intellectual property and copyright persists even after death. No amount of internet inactivity will change that, the only way for copyright to decade is 70 years after the holder death iirc. 

Uploading mods to curseforge won't change the licensing, the only thing a developer agrees to by uploading on their site is allowing the artifact (jar or whatever the uploaded content is) to be hosted and publicly available from there.
It changes absolutely nothing about the original licensing or the source code, or the rights of a third party like this

**Comment by Ruchian** [comment]
Yes, if it can be confirmed it was made by a single human entity.
The problem now is that. Emoniph on github could very well be someone else. Someone who didn't made the mod. The Emoniph on twitter, facebook, may just be fans. You see, the licence is linked to a username and that username doesn't direct to a specific person, but rather 'someone'. If I were to make an account on some platform, use it to email MsRandom and say 'you got my permission, signed by Emoniph', it voids the entire thing it is supports to protect.

**Comment by zabi94** [comment]
Except there are plenty of means for him to communicate with us. Like the witchery website

**Comment by Ruchian** [comment]
Your last message is completely true. No longer exists, dead, is a saying on the internet when a username stops replying posts or after inactivity for too long. It's because people don't actually talk to someone, but more or less leave messages and hope to read something later. It's kinda a meme thing.

I looked it up for you. Apparently in the US, once you're dead (unable to update your licence or agreements), it lasts for 28 years since 1998. (However, if you have a system set up that renews the licence, then indeed it lasts 70 years after death and if the work was published anonymously and renewed, it lasts for 95 years). Oddly enough the original copyright holder has the right to revoke adjustments to the licence he made for 35 years after the initial adjustment.
 
If you're talking about silence on the internet, You could also say legally 28 years unless the licence is automatically renewed.

There is a thing apparently that gives people automatic copyright protection. The thing is though, there are a couple of problems apparently. One of them being the thing I just mentioned. You need some legal way of proving that you (the creator) are you and a way to prove what of 'that' is you, basically. Otherwise its not goiing to work very well. To receive such proper copyright protection under the US law, you need to register your work in full and keep track of specific records, such as distribution, etc. (see copyright.gov) There is also a difference between copyright and patent, which isn't really apparent in most licences. Patent is basically the right to distribute a work. This must be registered for it to count since its not automatically protected, meaning other sites mirroring a download that isn't something you can guard against and its already hard to do on the internet.

According to Twitch's licence, when you put something on their platform, they receive the rights to do whatever they want with it. The same happens for anything they received when it was uploaded to Curse under their licence, however a difference is that the rights are revoked when the content is removed by the uploader.

So, summarized: I was wrong before. ;p
though that problem I mentioned there still exists unfortunately. But copyright aside.
The question is if it is ethical.

Is it okay to update a closed sourced mod that was made for entertainment to a later minecraft version? 
I mentioned in my post that got downvoted that, since it was made for the community, I don't see a problem to update the mod, so long the original author is kept up with it.
Despite it going against the terms stated on his site.
 and the reason for that is because the mod is made as entertainment for the community. If they cannot enjoy the said entertainment, it's just plain stupid.
100 years later, people won't play minecraft 1.7.10. It's not a painting of code you can just see.
Heck, his work distributed is closed source, meaning officially you're not allowed to decompile and look at its inner works either. (publicly at least)
meh-

I hope emoniph drops by once at least and think 'you what mate?' or something. even a complaint is still a message. If he doesn't want to mod anymore, then throw the code at the public or something. Currently its basically automaticallly going into the garbage bin this way.

I also wonder if this subject started on the Bewitchment mod community as well, since that was also a port attempt, but you're right. It's not disclosing emoniph's sourcecode without his permission.

**Comment by Gamebuster19901** [comment]
Copyright and patents are completely different things.

Patents protect inventions and methods of operation. 

Copyright protects works of art/expression.

**Comment by Gamebuster19901** [comment]
So your statement saying that the work isn't protected because it isn't registered is incorrect.

Patents must be registered to be protected, but Copyrights are protected at the moment it appears in a tangible medium of expression, which usually happens the moment a work is created.

**Comment by Ruchian** [comment]
Hm, I derailed the subject. I apologize. I also apologize for replying.
There is something I don't like myself, and its when people don't really read what I said, but I'll ignore it.

I understand completely why you wouldn't like it when someone would port your work, but there is a condition you need to consider here, and that condition is the wait.
If you're not even logging in on github for 4 years, would you care? Maybe that is too cruel.
If you have a program you haven't worked on, 4 years later someone decompiled it and decided to update it so that it can be used on the newer OS, would you mind?
Of course you would mind if it happens immediately after you release it (or within a month maybe), but in four years you can easily forget you even made something like that.
Yes, you would still mind if you intended to do it yourself, but considering the wait and no messages, I doubt emoniph would. Yes, I would mind, if I was already doing that since someone took over the job and make me wonder why I even started it. I can understand why you would, but there are two sides of the coin. You as developer and the fanbase you released it for.

True, breaking the licence is never good. I can't agree with it at all, but I wonder if the licence in this case was correct. True, it doesn't matter, as licences are as is and people should just respect them, but I still wonder. I mean, forge changed completely in the mean time. 1.17.10 is 2014, 1.12.2 is 2017.
1.15.1 is december 2019.
Currently this piece of software cannot be used unless individuals downgrade their minecraft client.
Have you seen the number of comments and requests at various locations in the mean time? That's basically what I feel sorry for. I think how much it damaged this bulk of mass (the people) is greater than it hurts emoniph to port it up at this point and on the other hand, like restoring a painting to its original feel for the modern years, it seems to me a way to preserve the creative work and respect for the original author if done correctly. But perhaps in programming that way of thinking is wrong.

No, I don't program myself. But I have seen how some developers, even well recognized companies handle things. They pretty much copy paste everything, change the names of each function, value, file, etc. and they bypass copyright that way. I always wonder how ethical that is, but never mind I guess.
> Fortunately it seems that the dev might decide to start converting piece by piece to an original work, which is still not ideal but fine with me.
that basically. It's not possible to do this with models and such, but yeah.
(also I have seen other mods in their mod history basically having files belonging to other mods)
Technically publishing those original files in your repository breaks copyright, but apparently only temporarily.

**Comment by cpoppy64** [comment]
Can you copyright an open-source mod for an open-source game?

**Comment by CammiePone** [comment]
> 
> 
> Can you copyright an open-source mod for an open-source game?

Yes, you can. But just fyi, Minecraft is closed-source, not open-source. You can copyright just about any work of art. Attempting to say you can't copyright code because it's open-source is like saying that Bob Ross can't copyright his paintings because he did them all on camera and showed others how to do it themselves. If someone stole his paintings and claimed them as their own work of art, or even if they bought them and claimed them as their art, they could get sued.
